controller/metrics/experiments.go

```diff
@@ -56,21 +38,18 @@ func (c *experimentCollector) Collect(ch chan<- prometheus.Metric) {
 }
 
 func collectExperiments(ch chan<- prometheus.Metric, ex *v1alpha1.Experiment) {
-
-	addConstMetric := func(desc *prometheus.Desc, t prometheus.ValueType, v float64, lv ...string) {
-		lv = append([]string{ex.Namespace, ex.Name}, lv...)
-		ch <- prometheus.MustNewConstMetric(desc, t, v, lv...)
-	}
 	addGauge := func(desc *prometheus.Desc, v float64, lv ...string) {
-		addConstMetric(desc, prometheus.GaugeValue, v, lv...)
+		lv = append([]string{ex.Namespace, ex.Name}, lv...)
+		ch <- prometheus.MustNewConstMetric(desc, prometheus.GaugeValue, v, lv...)
 	}
+	calculatedPhase := ex.Status.Phase
 
-	addGauge(descExperimentInfo, 1)
+	addGauge(MetricExperimentInfo, 1, string(calculatedPhase))
 
-	calculatedPhase := ex.Status.Phase
-	addGauge(descExperimentPhase, boolFloat64(calculatedPhase == v1alpha1.AnalysisPhasePending || calculatedPhase == ""), string("Pending"))
-	addGauge(descExperimentPhase, boolFloat64(calculatedPhase == v1alpha1.AnalysisPhaseError), string("Error"))
-	addGauge(descExperimentPhase, boolFloat64(calculatedPhase == v1alpha1.AnalysisPhaseSuccessful), string("Successful"))
-	addGauge(descExperimentPhase, boolFloat64(calculatedPhase == v1alpha1.AnalysisPhaseRunning), string("Running"))
-	addGauge(descExperimentPhase, boolFloat64(calculatedPhase == v1alpha1.AnalysisPhaseInconclusive), string("Inconclusive"))
+	// DEPRECATED
```

Comment:
looks good overall..one comment i have is i see deprecated like  these for metrics..is there a reason why we say deprecated and if so what is the alternative and why do we need to still maintain these if deprecated.

Reply:
Dashboards (including our own) are still built on these deprecated metrics. So there will be a period of time when not all clusters are migrated to the new metrics. Until then, you need a release which has both metrics so that when 100% of your rollout installations are producing the new metrics, then you can update the dashboard.